### DISCUSSION TITLE
How to decelerate when in flight

### DISCUSSION THREAD

**Discussion by theorynice** [JSBSim-Team/jsbsim#612] [General]
Hello, everyone! I have a question and it will be appreciated if you help me with this. How to decelerate when in flight? Is there a special control variable to realize speed cut when in flight? Or only can decelerate by setting throttle cmd to 0 when in flight?

**Comment by dpculp** [comment]
The most common way is to increase drag on the airplane.  This is easily 
done by using a speedbrake.  To create a speedbrake for your model you 
first define the speedbrake's effect in the <DRAG> section of your 
configuration file.  Something like this:

```xml
     <function name="aero/coefficient/CDsb">
        <description>Drag_due_to_speedbrakes</description>
          <product>
            <property>aero/qbar-psf</property>
            <property>metrics/Sw-sqft</property>
            <property>fcs/speedbrake-pos-norm</property>
            <value>0.03</value>
          </product>
     </function>
```

This assumes you have created a flight control component for your speed 
brake, like this:

```xml
    <kinematic name="Speedbrake Control">
      <input>fcs/speedbrake-cmd-norm</input>
      <traverse>
        <setting>
           <position> 0 </position>
           <time>     0 </time>
        </setting>
        <setting>
           <position> 1 </position>
           <time>     1 </time>
        </setting>
      </traverse>
      <output>fcs/speedbrake-pos-norm</output>
    </kinematic>
```

This gives the speedbrake a one second transit time from fully open to 
fully retracted, which is a reasonable number.

You then control the speedbrake through the variable 
"fcs/speedbrake-cmd-norm".


-- Dave

**Reply by theorynice** [reply]
Thanks for your help. It greatly helps me. According to your suggestion, I can control ‘’fcs/speedbrake-pos-norm” to realize the sharp deceleration of f16 when in flight.

**Reply by seanmcleod** [reply]
The JSBSim F-16 model does model a speedbrake, which affects both the drag and the lift. Look through the F-16 model file for speedbrake.

**Comment by seanmcleod** [comment]
> This is easily done by using a speedbrake. 

Maybe if you're flying an airliner, fighter etc. as opposed to a general aviation plane 😉 

In which case you have two controls, the elevator and the throttle. You can increase drag by increasing the induced drag by using the elevator.

For example take a look at the trim results for the C172p in JSBSim, at 8,000ft level flight (gamma = 0). Note the change in alpha, elevator and throttle. Also note how below 70kt to continue flying slower in level flight the throttle actually has to increase. You've reached the backside of the power curve.

```sh
C172P
100kt
  Trim Results: 
       Angle of Attack:   0.39  wdot: -2.14e-04 Tolerance: 1.000000e-03  Passed
              Throttle:   0.89  udot: -9.75e-06 Tolerance: 1.000000e-03  Passed
            Pitch Trim:   0.19  qdot:  3.72e-08 Tolerance: 1.000000e-04  Passed

90kt
  Trim Results: 
       Angle of Attack:   1.29  wdot: -1.42e-04 Tolerance: 1.000000e-03  Passed
              Throttle:   0.82  udot: -1.43e-05 Tolerance: 1.000000e-03  Passed
            Pitch Trim:   0.13  qdot:  1.98e-08 Tolerance: 1.000000e-04  Passed

80kt
  Trim Results: 
       Angle of Attack:   2.54  wdot: -5.10e-04 Tolerance: 1.000000e-03  Passed
              Throttle:   0.77  udot: -4.15e-05 Tolerance: 1.000000e-03  Passed
            Pitch Trim:   0.05  qdot:  5.83e-08 Tolerance: 1.000000e-04  Passed

75kt
  Trim Results: 
       Angle of Attack:   3.35  wdot: -2.48e-04 Tolerance: 1.000000e-03  Passed
              Throttle:   0.76  udot: -2.36e-05 Tolerance: 1.000000e-03  Passed
            Pitch Trim:   0.00  qdot: -5.50e-08 Tolerance: 1.000000e-04  Passed

70kt
  Trim Results: 
       Angle of Attack:   4.33  wdot:  1.20e-04 Tolerance: 1.000000e-03  Passed
              Throttle:   0.76  udot: -4.11e-05 Tolerance: 1.000000e-03  Passed
            Pitch Trim:  -0.04  qdot: -1.10e-07 Tolerance: 1.000000e-04  Passed

65kt
  Trim Results: 
       Angle of Attack:   5.35  wdot:  6.93e-04 Tolerance: 1.000000e-03  Passed
              Throttle:   0.77  udot: -1.56e-04 Tolerance: 1.000000e-03  Passed
            Pitch Trim:  -0.08  qdot: -1.48e-06 Tolerance: 1.000000e-04  Passed

60kt
  Trim Results: 
       Angle of Attack:   6.62  wdot:  2.54e-04 Tolerance: 1.000000e-03  Passed
              Throttle:   0.78  udot: -3.30e-05 Tolerance: 1.000000e-03  Passed
            Pitch Trim:  -0.13  qdot: -1.92e-07 Tolerance: 1.000000e-04  Passed
```

Brief example from the 737 model, also at 8,000ft level flight. So the throttle stays the same but you can bleed off 10kt by increasing elevator angle, increase alpha and keep things in trim.

```sh
737
200kt
  Trim Results: 
       Angle of Attack:   6.52  wdot:  1.07e-04 Tolerance: 1.000000e-03  Passed
              Throttle:   0.62  udot: -9.23e-05 Tolerance: 1.000000e-03  Passed
            Pitch Trim:  -0.42  qdot: -1.48e-10 Tolerance: 1.000000e-04  Passed

190kt
  Trim Results: 
       Angle of Attack:   7.50  wdot:  1.19e-04 Tolerance: 1.000000e-03  Passed
              Throttle:   0.62  udot: -1.04e-04 Tolerance: 1.000000e-03  Passed
            Pitch Trim:  -0.49  qdot: -1.64e-10 Tolerance: 1.000000e-04  Passed
```

If you are going to add a speedbrake to your model don't forget that in addition to increasing drag they generally also reduce lift, e.g. from the JSBSim 737 model.

```xml
        <function name="aero/function/kCLsb">
            <description>Change_in_lift_due_to_speed_brake</description>
            <table>
                <independentVar>fcs/speedbrake-pos-norm</independentVar>
                <tableData>
                    0.0000	1.0
                    0.1000	0.85
                </tableData>
            </table>
        </function>
```

```xml
            <function name="aero/coefficient/CLalpha">
                <description>Lift_due_to_alpha</description>
                <product>
                    <property>aero/qbar-psf</property>
                    <property>metrics/Sw-sqft</property>
                    <property>aero/function/kCLge</property>
                    <property>aero/function/kCLsb</property>
                    <property>aero/function/kCLsp</property>
                      <table>
                          <independentVar>aero/alpha-rad</independentVar>
                          <tableData>
                             -0.20     -0.68
                              0.00	0.20
                              0.23	1.20
                              0.46	0.20
                          </tableData>
                      </table>
                </product>
            </function>
```

> Is there a special control variable to realize speed cut when in flight?

So no, in general there is no single control/inceptor, it's a balancing act between multiple inceptors (elevator, throttle, speed brake), unless you have a FCS which does give you a single command option, but behind the scenes the FCS will be controlling multiple inceptors to achieve the command.

**Comment by seanmcleod** [comment]
In graphical form, results from JSBSim trim routine for 8,000ft, level flight (gamma=0) for speeds ranging from 100KIAS to 45KIAS, showing the change in alpha and the change in inceptor positions.

![image](https://user-images.githubusercontent.com/824856/163707570-0e074359-5fa8-49a3-8d3b-d747dfb128e9.png)

And for the 737 covering 350KIAS to 160KIAS.

![image](https://user-images.githubusercontent.com/824856/163707756-aee76027-b794-418a-a669-b7c5436682c0.png)

**Reply by theorynice** [reply]
I appreciate your demo and have learned a lot from this. Thanks very much. By the way, does the 'forces/lod-norm' represent the normal load factor and implicit the agility of maneuver?

**Comment by seanmcleod** [comment]
In general if you're not sure what a particular property represents, e.g. `forces/lod-norm` look through the JSBSim source code, e.g.

https://github.com/JSBSim-Team/jsbsim/blob/aa2f1276c2d117865da448f20a7dff4e8c4bb025/src/models/FGAerodynamics.cpp#L624

Then look up `GetLoD()`

https://github.com/JSBSim-Team/jsbsim/blob/375f5be0fdfeb210f4cf66f938e3e6fd426c2966/src/models/FGAerodynamics.h#L207-L208

**Reply by theorynice** [reply]
Thank you very much. There is a new problem confusing me as the below figures show:
![Figure_3q](https://user-images.githubusercontent.com/39797845/163972107-b474fdc5-2477-474a-8a9b-8ae545830ac4.png)
![Figure_4q](https://user-images.githubusercontent.com/39797845/163972119-6c52fcc4-c151-4689-8995-d32888555ca1.png)
It is a task that let f15 turn to a target heading under a target velocity and altitude. As the figures show when the throttle-cmd-norm was set to 1, why the velocity keeps nearly invariant rather than increase?

**Comment by seanmcleod** [comment]
Remember, flight dynamics generally isn't 1-dimensional. So simply increasing the throttle doesn't guarantee a 1-1 mapping to speed. Look at what else is happening, you've pulled back on the stick so the elevator is at maximum deflection which will increase alpha and g, which ramps up the induced drag. So in addition to profile drag the thrust also has to overcome the induced drag if you want the aircraft to accelerate.

**Comment by seanmcleod** [comment]
In terms of fighter maneuverability and energy look up Energy Maneuverability diagrams, often also called dog-house plots. Here is a simplified one which only shows the Ps = 0 curve in green.

So for the given conditions (19,000ft, weight and drag index and full thrust) the Ps = 0 curve represents a net energy of 0, so the aircraft could sustain any point on the green line, i.e. it wouldn't accelerate or decelerate and it would achieve the given performance in terms of speed, turn-rate, turn-radius and g.

However above the green line there would be a net loss of energy, which means either the aircraft would decelerate or it would have to descend it if wanted to keep the same speed.

Below the green line there is excess energy, so the aircraft could maintain (turn-rate, turn-radius, g) and could then use the excess energy to either increase speed or to climb.

![image](https://user-images.githubusercontent.com/824856/163977853-0bb42cb9-882f-4bbf-8d7c-f1982975f919.png)

**Reply by theorynice** [reply]
Wonderful explanation, there is too much to learn for me! Thank you.
